clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystemCore.java

```diff
@@ -166,6 +167,15 @@ TachyonFile openIfExists(TachyonURI path, OpenOptions options) throws IOExceptio
       TachyonException;
 
   /**
+   * Persists a {@link TachyonFile} to the UnderFileSystem.
+   *
+   * @param file the TachyonFile to persist
+   * @return true if the persist succeeded, false otherwise
+   */
+  boolean persistFile(TachyonFile file, PersistOptions options)
```

Comment:
The `persistFile` method on the master will fail if you pass it a directory. I didn't want to deal with directories, because I wanted `TfsShell` to handle exceptions differently than we would normally do. Namely, it tries persisting all the files in the directory, even if some of them fail, and then it combines all the failed files into one exception. I thought that would be weird to implement in a non-TfsShell method, because we only do that in TfsShell.